miniDAS/format.py

```diff
@@ -0,0 +1,161 @@
+from __future__ import annotations
+
+import logging
+from dataclasses import asdict, dataclass, field
+from pathlib import Path
+from typing import Any, Literal
+
+import h5py
+import numpy as np
+
+DataUnit = Literal["m/m", "cm/m", "mm/m"]
+ScaledUnit = Literal["ue/m"]
+_ALLOWED_DTYPES = (np.int32, np.float32)
+
+
+class OutOfBoundsError(BaseException):
+    ...
+
+
+@dataclass
+class Meta:
+    data_unit: DataUnit
+    start_time_ns: int
+    scale_factor: float
+    units_after_scaling: ScaledUnit
+    sampling_rate: int
+    gauge_length: float
+    latitudes: list[float]
+    longitudes: list[float]
+    elevations: list[float]
+    end_time_ns: int = 0
+
+    @property
+    def deltat(self) -> float:
+        return 1.0 / self.sampling_rate
+
+    @property
+    def start_time(self) -> float:
+        return self.start_time_ns / 1e9
+
+    @property
+    def end_time(self) -> float:
+        return self.end_time_ns / 1e9
+
+
+class miniDAS:
+    _file: h5py.File | None
+
+    version: int = 1
```

Comment:
I thought that version 1.0.0 was the agreed approach to versioning afterall?
At least a 1.0 would be appropriate

Reply:
File formats should be non-breaking between minor versions. It is industry standard to use only major... see e.g. hdf5 vs hdf4

Reply:
yes, agreed.
But here we only have version 1, as an integer
What about all the discussion we had earlier about versioning?